ch_pipeline/hfb/flagging.py

```diff
@@ -0,0 +1,145 @@
+"""HFB Tasks for flagging data
+"""
+import numpy as np
+
+from caput import config
+
+from draco.core import task
+
+from containers import HFBRFIMask
+
+
+class HFBRNTMask(task.SingleTask):
+    """Identify RFI in HFB data using the radiometer noise test, averaging over beams.
+
+    Attributes
+    ----------
+    threshold : float
+        The desired threshold for the RFI fliter. Data with sensitivity metric
+        values above this thershold will be considered RFI.
+    keep_sens : bool
+        Save the sensitivity metric data that were used to construct
+        the mask in the output container.
+    """
+
+    threshold = config.Property(proptype=float, default=1.6)
+    keep_sens = config.Property(proptype=bool, default=False)
+
+    def process(self, stream):
+        """Derive an RFI mask.
+
+        Parameters
+        ----------
+        stream : containers.HFBData
+            Container with HFB data and weights.
+
+        Returns
+        -------
+        out : containers.HFBRFIMask
+            Boolean mask that can be applied to an HFB data container
+            with the task `ApplyHFBMask` to mask contaminated
+            frequencies, subfrequencies and time samples.
+        """
+
+        # Extract data and weight arrays, averaging over beams
+        data = np.mean(stream.hfb[:], axis=2)
+        weight = np.mean(stream.weight[:], axis=2)
+
+        # Number of samples per data point in the HFB data:
+        # N_S = Delta nu * Delta t, where Delta nu is the frequency resolution
+        # (390.625 kHz / 128) and Delta t the integration time (10 s)
+        # NOTE: Should this be extracted from the container attributes? In that
+        # case this information needs to be added to the attributes upstream.
+        N_SAMP = 390625.0 / 128.0 * 10.0
+
+        # Ideal radiometer equation
+        radiometer = data**2 / N_SAMP
```

Comment:
I believe we also store a `frac_lost` dataset. That's important to use in case we have lost data upstream, and so the sample does not contain as much integration time as expected.

Reply:
Added.